### DISCUSSION TITLE
"Your kernel headers for kernel 5.4.72-microsoft-standard-WSL2 cannot be found."

### DISCUSSION THREAD

**Discussion by 0x14307** [microsoft/WSL#6650] [General]
I'm trying to install a package with dpkg and I get this error:
"Error! Your kernel headers for kernel 5.4.72-microsoft-standard-WSL2 cannot be found.
Please install the linux-headers-5.4.72-microsoft-standard-WSL2 package"

How can I fix this?

**Comment by gabefair** [comment]
Can you tell us which package or information about the software you are trying to install? For example if it is a 10g lan driver that can help us narrow down the issue.

**Reply by 0x14307** [reply]
I'm trying to install Darling-DKMS.

**Comment by 0x14307** [comment]
I'm trying to install Darling-DKMS.

**Comment by awaldfogel** [comment]
Any update on kernel headers? I'm trying to install a MOXA serial driver whose build relies on exact matches for kernel headers. It's looking for kernel 5.15.79.1-microsoft-standard-WSL2 Is there an alternate repository where I can get it? Otherwise I'm stuck with an unsupported tweak on their install directory if I can get the driver to build with the generic headers.  Zero interest in building my own kernels!

**Reply by zcobol** [reply]
You could clone https://github.com/microsoft/WSL2-Linux-Kernel.git and checkout `linux-msft-wsl-5.15.79.1`. Export the headers and build your module.

**Reply by inliquid** [reply]
Thank you now initial message disappeared and there is new thing:
```
fatal error: 'generated/autoconf.h' file not found
```

**Reply by TishSerg** [reply]
> fatal error: 'generated/autoconf.h' file not found

I have this problem too. Why Microsoft not willing to share fully functional linux-headers for WSL kernel?

**Comment by awaldfogel** [comment]
Thanks! Was able to build the driver but then modprobe won't let me install it under Docker (which I guess makes sense). Will WSL2 let me modprobe install a driver module in the base image? If it installs the device appears as /dev/ttyrxx. Will those ttyrxx devices then appear in the docker images?  I'm new to this. 

Trying to consolidate a bunch of services on a single machine. One service has a special requirement to access a remote serial device
